core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +265,95 @@ public void setConf(Configuration conf) {
+          }
+          return sql.executeUpdate();
+        }
+      });
+
+      if (insertedRecords == properties.size()) {
+        LOG.debug("Successfully inserted {} properties for namespace {}", properties, namespaceName);
+        return true;
+      } else {
+        throw new IllegalStateException();
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e, "Interrupted in call to insertProperties(namespace, properties) " +
+          "Namespace: %s", namespace);
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to insertProperties to namespace: %s in catalog: %s", namespace,
+          catalogName);
+    }
+  }
+
+  private boolean updateProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int updatedRecords = connections.run(conn -> {
+        String sqlStatement = JdbcUtil.updatePropertiesStatement(properties.size());
+
+        try (PreparedStatement sql = conn.prepareStatement(sqlStatement)) {
+          int rowIndex = 0;
+          for (Map.Entry<String, String> keyValue : properties.entrySet()) {
+            sql.setString(rowIndex + 1, keyValue.getKey());
+            sql.setString(rowIndex + 2, keyValue.getValue());
+            rowIndex += 2;
+          }
+          sql.setString(rowIndex + 1, catalogName);
+          sql.setString(rowIndex + 2, namespaceName);
+          rowIndex += 2;
+          for (String key : properties.keySet()) {
+            sql.setString(rowIndex + 1, key);
+            rowIndex += 1;
+          }
+          LOG.info("Final log string {}", sql);
+          return sql.executeUpdate();
+        }
+      });
+
+      if (updatedRecords == properties.size()) {
+        LOG.debug("Successfully updated {} to new namespace: {}", properties, namespaceName);
+        return true;
+      } else {
+        throw new IllegalStateException();
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e,
+          "Interrupted in call to updateProperties(namespace, properties) Namespace: %s", namespace);
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to updateProperties to namespace: %s in catalog: %s", namespace,
+              catalogName);
+    }
+  }
+
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
-    throw new UnsupportedOperationException("Cannot create namespace " + namespace +
-        ": createNamespace is not supported");
+    if (namespaceExists(namespace)) {
+      throw new AlreadyExistsException("Namespace already exists: %s", namespace);
+    }
+
+    if (metadata == null || metadata.isEmpty()) {
+      throw new IllegalArgumentException("Cannot create a namespace with null or empty metadata");
+    }
```

Comment:
Found the original review comment: https://github.com/apache/iceberg/pull/3275#discussion_r735009192 about null metadata. I added the check based on that and for empty metadata.
@rdblue: I agree with @kbendick  about adding a property like `creation timestamp` with the metadata if not provided. Let me know if you think otherwise.

Reply:
I think its fine the way it is to throw if people try to create a namespace without properties.

However, to provide information about why (without adding comments), I'd use
```java
ValidationException.check(metadata != null && !metadata.isEmpty(), "Cannot create a namespace without metadata in the JDBC catalog as the JDBC catalog uses the namespace metadata as an existence check");
```

It would also be ok to just put a comment above it about the existence check.

We can add a property on behalf of the user in a follow up PR.